### PR TITLE
py3 hotness

### PR DESCRIPTION
**Pull request by bcen**
<del>TODO:</del> No more todos, the future is now.

- [x] Wheel & test it on hotlanta
- [x] Use `future` and test it on hotlanta

### CODE REVIEW

**Comment by irmiller22** [comment]
@bcen let me know when you need someone to review - happy to help out here. additionally, do you think it's worthwhile to move from travisci to cci?

**Comment by bcen** [comment]
Thanks, sure @irmiller22 moar eyes the better.

**Comment by lapointexavier** [comment]
👀

**Comment by kevinbirch** [comment]
> do you think it's worthwhile to move from travisci to cci

that would be great!

**Comment by bcen** [comment]
> do you think it's worthwhile to move from travisci to cci

Sure thing, cool to be on a different PR?

**Comment by irmiller22** [comment]
@bcen yep, ill do it in a diff pr once CCI comes back online from its outage

**Comment by bcen** [comment]
@lzhou89 @kevinbirch @irmiller22 
This is ready for final review. jsonmatch 0.3 on hotlanta also passed all its test.

**Comment by lzhou89** [comment]
> Not really, JOB is still the original author. We can add other ppl as maintainers tho.

but does his email still work? 😓 we're listing `author_email='jamesob@percolate.com',`

**Comment by bcen** [comment]
@lra any suggestion which email we can set to as author? I mean we still want to give credit to JOB as the original author, but does that email still work?

**Comment by bcen** [comment]
@lzhou89 @kevinbirch using `future` now. a9858fa

File: jsonmatch/matcher.py [hunk 1]
```diff
@@ -137,7 +143,9 @@ def _find_breaks(self, exp_d, test_d, key_trail, is_ordered, breaks):
             log.info("Someone is missing keys.")
             return breaks
 
-        for key, val in exp_d.items():
+        # PY3: `for key, val in exp_d.items()`
+        for key, val in iteritems(exp_d):
```
Comment (lzhou89) [inline jsonmatch/matcher.py]: if you `from builtins import dict` you can directly do `exp_d.items()`.  Though it is kind of inefficient in py2... 😓

File: jsonmatch/matcher.py [hunk 2]
```diff
@@ -224,9 +233,11 @@ def __init__(self, spec, against):
     def add_break(self, path_tuple, spec_val, against_val):
         self.paths_to_breaks[path_tuple] = (spec_val, against_val)
 
-    def __nonzero__(self):
+    def __bool__(self):
         return bool(self.paths_to_breaks)
 
+    __nonzero__ = __bool__
```
Comment (lzhou89) [inline jsonmatch/matcher.py]: if you `from builtins import object` you don't need to define `__nonzero__`

File: jsonmatch/matcher.py (hunk 1 above)
Comment (bcen) [reply jsonmatch/matcher.py]: I'll leave this as it is, unless others disagree. While hotlanta has some large dicts but I don't think it matters much either way (just a gut check, no benchmark).

File: jsonmatch/matcher.py (hunk 2 above)
Comment (bcen) [reply jsonmatch/matcher.py]: Updated

**Comment by lzhou89** [comment]
lgtm.  @irmiller22 @kevinbirch any other comments?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## jsonmatch 0.3 / 2018-TODO
```
Comment (bcen) [inline CHANGELOG.md]: Will update this to the date of approval/merge.

**Comment by bcen** [comment]
Hoping to merge this next Monday (2018-07-30) and updated the date on changelog.

**Comment by lra** [comment]
Regarding the email, either @jamesob still wants to be the maintainer of it, and he should put his personal email, either he does not and we should put whoever wants to maintain it from now on.

**Comment by jamesob** [comment]
Feel free to change the email, y'all. :heart:

**Comment by lra** [comment]
@bcen Can you push 0.2 and 0.3 on https://pypi.org/project/jsonmatch/ ? Still 0.1 there.

**Comment by bcen** [comment]
@lra can you add me to the jsonmatch project on pypi. thanks.

**Comment by lra** [comment]
I'm not part of the project:
screen shot 2018-08-01 at 3 10 53 pm

@jamesob can you?

**Comment by jamesob** [comment]
Done. Should be all good, @bcen.

**Comment by bcen** [comment]
Thank you @jamesob !!

**Pull request merged**